CMakeLists.txt

```diff
@@ -13,7 +13,15 @@ option(BUILD_GAME_LIBRARIES "Build game module libraries" ON)
 option(BUILD_GAME_QVMS "Build game module qvms" ON)
 option(BUILD_STANDALONE "Build binaries for standalone games" OFF)
 
-option(USE_ARCHLESS_FILENAMES "Don't include the architecture in binary filenames" OFF)
+if(APPLE)
```

Comment:
I realise you probably wanted to centralize all Apple-specific logic into `cmake/platforms/macos.cmake`, but I don't really see a way around this: if an option is going to have a platform-specific default, then the build system needs to figure out its default before defining the option.